src/controller/CHIPDeviceControllerSystemState.h

```diff
@@ -168,7 +170,7 @@ class DeviceControllerSystemState
         return mSystemLayer != nullptr && mUDPEndPointManager != nullptr && mTransportMgr != nullptr && mSessionMgr != nullptr &&
             mUnsolicitedStatusHandler != nullptr && mExchangeMgr != nullptr && mMessageCounterManager != nullptr &&
             mFabrics != nullptr && mCASESessionManager != nullptr && mSessionSetupPool != nullptr && mCASEClientPool != nullptr &&
-            mGroupDataProvider != nullptr && mSessionKeystore != nullptr;
+            mGroupDataProvider != nullptr && mReportScheduler != nullptr && mSessionKeystore != nullptr;
```

Comment:
This is an API-breaking change, no?  Things that used to work will now stop working unless someone goes and injects a report scheduler _for a situation that probably never has any ReadHandlers at all_.

This is not really workable.  It _must_ be possible to initialize a CHIPDeviceController without a scheduler.  If we need one for ReadHandlers, then either creation of those should fail if we don't have one or we should use a default one if none is provided.  But we can't just fail controller init like this.

Note that this is 100% causing your Darwin CI failures.